### PR TITLE
JDP230502-19 - CartRepositoryTestSuite

### PR DESCRIPTION
**Pull request by sew-k**
- CartRepositoryTestSuite - tests for Cart entity
- minor modifications of "cascade" parameter in fields annotattions (relations between entities)

### CODE REVIEW

File: src/test/java/com/kodilla/ecommercee/domain/CartRepositoryTestSuite.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import java.util.List;
+import java.util.Optional;
+
+@SpringBootTest
+@ExtendWith(SpringExtension.class)
+public class CartRepositoryTestSuite {
+    @Autowired
+    CartRepository repository;
+
+    List<CartProducts> cartProducts = new ArrayList<>();
```
Comment (WorstWorm) [inline src/test/java/com/kodilla/ecommercee/domain/CartRepositoryTestSuite.java]: I believe that a separate list should be assigned to each Cart - I suggest adding new ArrayList<CartProducts> as the last parameter to each of the Carts below.

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -1,4 +1,5 @@
-spring.datasource.url=jdbc:mysql://localhost:3306/kodilla_project?serverTimezone=Europe/Warsaw&useSSL=False
+spring.datasource.url=jdbc:mysql://localhost:3306/kodilla_project?serverTimezone=Europe/Warsaw&useSSL=False&\
```
Comment (WorstWorm) [inline src/main/resources/application.properties]: Here I suggest to revert back to the original version from master - the change doesn't seem to be needed for passing tests.

**Review by WorstWorm — COMMENTED** [review]
In order to pass the tests correctly, you need to add @Setter in the User and Order classes (along with imports). After that, everything should work fine, but I don't want to mess up modifying your branch myself, so you'd still need to verify with Circle.

File: src/test/java/com/kodilla/ecommercee/domain/CartRepositoryTestSuite.java (hunk 1 above)
Comment (sew-k) [reply src/test/java/com/kodilla/ecommercee/domain/CartRepositoryTestSuite.java]: Yes, you're right. This is my obvious mistake.

File: src/main/resources/application.properties (hunk 2 above)
Comment (sew-k) [reply src/main/resources/application.properties]: Not working. Last commit has all your suggestions.

**Review by WorstWorm — APPROVED** [review]
After a longer conversation - I agree that although the tests are not perfect in this form and do not exactly cover the problem, we need to include these changes into the master. Improving and validating tests (all) should become an additional task after all key classes will be ready.

**Review by JakubFir — APPROVED** [review]

**Review by annapawlas — APPROVED** [review]

**Pull request merged**